packtools/sps/validation/article_license.py

```diff
@@ -1,52 +1,191 @@
+                        'lang': 'en',
+                        'link': 'http://creativecommons.org/licenses/by/4.0/',
+                        'license_p': 'This is an article published in open access under a Creative Commons license.'
+                        },
+                    'got_value': {
+                        'lang': 'en',
+                        'link': 'http://creativecommons.org/licenses/by/4.0/',
+                        'license_p': 'This is an article published in open access under a Creative Commons license.'
+                        },
+                    'message': "Got {"
+                               "'lang': 'en', "
+                               "'link': 'http://creativecommons.org/licenses/by/4.0/', "
+                               "'license_p': 'This is an article published in open access under a Creative Commons license.'"
+                               "} expected {"
+                               "'lang': 'en', "
+                               "'link': 'http://creativecommons.org/licenses/by/4.0/', "
+                               "'license_p': 'This is an article published in open access under a Creative Commons license.'"
+                               "}",
+                    'advice': None
+                },
+                ...
+            ]
+        """
+        if not expected_value:
+            raise ValidationLicenseException("Provide a dictionary with license data for validation")
+
         obtained_value = self.article_license.licenses_by_lang
-        resp = {
-            "obtained_value": obtained_value,
-            "expected_value": expected_value
-        }
-        msg = []
-        result = 'ok'
-        for language, dictonary in obtained_value.items():
-            try:
-                if expected_value[language] == dictonary:
-                    msg.append(f'ok, the license text for {language} does match the license text adopted by the journal')
-                else:
-                    result = 'error'
-                    msg.append(f'error, the license text for {language} does not match the license text adopted by the journal')
-            except KeyError:
-                    result = 'error'
-                    msg.append(f'error, the language {language} is not foreseen by the journal')
 
-        resp.update({
-            "result": result,
-            "message": msg
-        })
-        return resp
+        for lang, data in obtained_value.items():
+            is_valid = expected_value.get(lang) == data
+            expected_value_msg = expected_value.get(
+                lang) if is_valid else 'License data that matches the language {}'.format(lang)
+            yield {
+                'title': 'Article license validation',
+                'xpath': './permissions//license',
+                'validation_type': 'value in list',
+                'response': 'OK' if is_valid else 'ERROR',
+                'expected_value': expected_value_msg,
```

Comment:
@Rossi-Luciano acho que percebi agora o que está estranho. Revise as demais classes de validação em que vc usa o for. A iteração tem que ser no expected e não no obtained.

Reply:
OK @robertatakenaka.